include/pisa/query/algorithm/wand_query.hpp

```diff
@@ -17,7 +17,7 @@ struct wand_query {
 
     template<typename CursorRange>
     uint64_t operator()(CursorRange &&cursors) {
-        using Cursor = typename CursorRange::value_type;
+        using Cursor = typename std::decay_t<CursorRange>::value_type;
```

Comment:
Can we change this for all the other algorithms?

Reply:
We should. We must.